DOCUMENTATION.md

```diff
@@ -734,6 +698,44 @@ If the request is not successful, the client returns an `HTTPError` containing t
+|Created|GOV.UK Notify has placed the message in a queue, ready to be sent to the provider. It should only remain in this state for a few seconds.|
+|Sending|GOV.UK Notify has sent the message to the provider. The provider will try to deliver the message to the recipient for up to 72 hours. GOV.UK Notify is waiting for delivery information.|
+|Delivered|The message was successfully delivered.|
+|Failed|This covers all failure statuses:<br>- `permanent-failure` - "The provider could not deliver the message because the email address was wrong. You should remove these email addresses from your database."<br>- `temporary-failure` - "The provider could not deliver the message. This can happen when the recipient’s inbox is full. You can try to send the message again."<br>- `technical-failure` - "Your message was not sent because there was a problem between Notify and the provider.<br>You’ll have to try sending your messages again."|
+
+## Text message status descriptions
+
+|Status|Description|
+|:---|:---|
+|Created|GOV.UK Notify has placed the message in a queue, ready to be sent to the provider. It should only remain in this state for a few seconds.|
+|Sending|GOV.UK Notify has sent the message to the provider. The provider will try to deliver the message to the recipient for up to 72 hours. GOV.UK Notify is waiting for delivery information.|
+|Pending|GOV.UK Notify is waiting for more delivery information.<br>GOV.UK Notify received a callback from the provider but the recipient’s device has not yet responded. Another callback from the provider determines the final status of the notification.|
+|Sent / Sent internationally|The message was sent to an international number. The mobile networks in some countries do not provide any more delivery information. The GOV.UK Notify client API returns this status as `sent`. The GOV.UK Notify client app returns this status as `Sent internationally`.|
+|Delivered|The message was successfully delivered.|
+|Failed|This covers all failure statuses:<br>- `permanent-failure` - "The provider could not deliver the message. This can happen if the phone number was wrong or if the network operator rejects the message. If you’re sure that these phone numbers are correct, you should [contact GOV.UK Notify support](https://www.notifications.service.gov.uk/support). If not, you should remove them from your database. You’ll still be charged for text messages that cannot be delivered."<br>- `temporary-failure` - "The provider could not deliver the message. This can happen when the recipient’s phone is off, has no signal, or their text message inbox is full. You can try to send the message again. You’ll still be charged for text messages to phones that are not accepting messages."<br>- `technical-failure` - "Your message was not sent because there was a problem between Notify and the provider.<br>You’ll have to try sending your messages again. You will not be charged for text messages that are affected by a technical failure."|
```

Comment:
@idavidmcdonald do we return `failed` as a status? Or is it always the more specific `permanent-failure`, `temporary-failure` and `technical-failure`?

Reply:
It's always more specific

Reply:
In that case I might split these out in the table, instead of grouping them all under 'Failed' in the status column. This moves us closer to being able to style the statuses as code without being inaccurate or misleading.

Reply:
we never return `failed` as a status. However, when fetching notifications, you can set `status='failed'` to return all failed notifications, regardless of whether they were permanent, technical or temporary. 

If you were building up a failed messages dashboard or report, you should use this otherwise you'd get, eg, "the 250 most recent temporary failures" and "the 250 most recent permanent failures" separately and the time spans of those two might not match, leading to confusing results.

At least three different services have used this search filter a total of ~1500 times in the last week

Reply:
@leohemsted That's really good to know.

As these tables are intended to explain each status – is there value in doing something like this?

|Status|Description|
|:---|:---|
|`failed`| This covers all 3 failure statuses: `permanent-failure`, `temporary-failure` and `technical-failure`|
|`permanent-failure`|The provider could not...|
|`temporary-failure`|The provider could not...|
|`technical-failure`|Your message was not...|